spdx-convertor/convert.php

```diff
@@ -0,0 +1,274 @@
+<?php
+
+declare(strict_types=1);
+/**
+ * SPDX-FileCopyrightText: 2024 Joas Schilling <coding@schilljs.com>
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+require_once __DIR__ . '/vendor/autoload.php';
+
+if ($argc < 2 || in_array('--help', $argv) || in_array('-h', $argv)) {
+	die("convert.php [--dry-run] <path>\n");
+}
+
+$isDryRun = $argv[1] === '--dry-run';
+$path = $isDryRun ? $argv[2] : $argv[1];
+
+function generateSpdxContent(string $originalHeader, string $file): array {
+	$newHeaderLines = [];
+	$authors = [];
+	$license = null;
+
+	foreach (explode("\n", $originalHeader) as $line) {
+		// @copyright Copyright (c) 2023 Joas Schilling <coding@schilljs.com>
```

Comment:
There's a minor issue with creation years in copyright headers:

- If several copyrights were in the file before, that leads to duplicated headers:
  ```
  /**
   * SPDX-FileCopyrightText: 2023 Nextcloud GmbH and Nextcloud contributors
   * SPDX-FileCopyrightText: 2022 Nextcloud GmbH and Nextcloud contributors
   * SPDX-License-Identifier: AGPL-3.0-or-later
   */
   ```
  We should keep only the earliest year in that case
- If copyright doesn't have a year listed, that leads to non-specified header:
  ```
  /**
   * SPDX-FileCopyrightText: Nextcloud GmbH and Nextcloud contributors
   * SPDX-License-Identifier: AGPL-3.0-or-later
   */
   ```
  We may extract the creation year from the first known commit to that file (shell script example):
  `creation_year=$(git log --diff-filter=A --follow --format=%aI "$file" | tail -n 1 | cut -d'-' -f1)`

Reply:
The ones without a year is where I added it manually

Reply:
Not sure, if that would be complex to automate for files without license as well

Reply:
I think its better to have a manual look when its missing.
But the duplicate one I will solve

Reply:
Done